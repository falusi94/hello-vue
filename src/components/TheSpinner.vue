<template>
  <div>
    <div class="spinner-container">
      <img class="spinner-image" src="../assets/spinner.png" alt="Loading"/>
    </div>
    <div class="spinner-text">
      Loading...
    </div>
  </div>
</template>

<script>
export default {
  name: 'TheSpinner',
};
</script>

<style lang="scss" scoped>
@keyframes loading {
  0%   { transform: rotate(0deg);   }
  100% { transform: rotate(360deg); }
}
.spinner {
  &-container {
    margin: 0 auto 0 auto;
    width: 50px;
  }
  &-image {
    animation: loading 1.8s infinite linear;
    width: 50px;
  }
  &-text {
    text-align: center;
    padding: 25px 0;
  }
}
</style>
