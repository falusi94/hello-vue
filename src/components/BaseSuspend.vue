<template>
  <div v-if=(isLoading)>
    <TheSpinner/>
  </div>
  <div v-else>
    <slot />
  </div>
</template>

<script>
import TheSpinner from './TheSpinner.vue';

export default {
  name: 'BaseSuspend',
  components: {
    TheSpinner,
  },
  props: {
    isLoading: Boolean,
  },
};
</script>
