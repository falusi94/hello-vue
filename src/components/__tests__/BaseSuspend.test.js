import { shallowMount, mount } from '@vue/test-utils';
import BaseSuspend from '../BaseSuspend.vue';
import TheSpinner from '../TheSpinner.vue';

const setup = (propsData = { isLoading: true }) => mount(BaseSuspend, {
  slots: {
    default: '<div>CONTENT</div>',
  },
  propsData,
});

describe('BaseSuspend.vue', () => {
  it('renders properly', () => {
    const wrapper = shallowMount(BaseSuspend);
    expect(wrapper).toMatchSnapshot();
  });

  describe('when isLoading is set to false', () => {
    it('shows the content', () => {
      const wrapper = setup({ isLoading: false });

      const spinner = wrapper.findComponent(TheSpinner);
      expect(spinner.exists()).toBe(false);
      expect(wrapper.text()).toContain('CONTENT');
    });
  });

  describe('when isLoading is set to true', () => {
    it('shows the spinner', () => {
      const wrapper = setup({ isLoading: true });

      const spinner = wrapper.findComponent(TheSpinner);
      expect(spinner.exists()).toBe(true);
      expect(wrapper.text()).not.toContain('CONTENT');
    });
  });
});
