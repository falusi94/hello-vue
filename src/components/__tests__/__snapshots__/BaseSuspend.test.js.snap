// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSuspend.vue renders properly 1`] = `<div></div>`;
